tests/ha/single_pod_restart/test_data_server_pod_restart_api.py

```diff
@@ -456,3 +456,185 @@ def test_obj_overwrite_during_data_server_pod_restart(self):
+                    "and services states are as expected & remaining pods status is online")
+        LOGGER.info("Step 3: Download the copied objects & verify etags.")
+        for bkt, obj in bkt_obj_dict.items():
+            resp = self.s3_test_obj.get_object(bucket=bkt, key=obj)
+            LOGGER.info("Get object response: %s", resp)
+            get_etag = resp[1]["ETag"]
+            assert_utils.assert_equal(put_etag, get_etag, "Failed in verification of Put & Get "
+                                                          f"Etag for object {obj} of bucket "
+                                                          f"{bkt}.")
+        LOGGER.info("Step 3: Downloaded copied objects & verify etags.")
+        bkt_obj_dict1 = bkt_obj_dict.copy()
+        t_t = int(perf_counter_ns())
+        bucket_name = self.bucket_name
+        bkt_op = False
+        if CMN_CFG["dtm0_disabled"]:
+            LOGGER.info("Create and list buckets")
+            bkt_obj_dict.clear()
+            for cnt in range(bkt_cnt):
+                bkt_obj_dict[f"ha-bkt{cnt}-{t_t}"] = f"ha-obj{cnt}-{t_t}"
+            bucket_name = f"ha-mp-bkt-{int(perf_counter_ns())}-1"
+            bkt_op = True
+        else:
+            for idx, bkt in enumerate(bkt_obj_dict):
+                bkt_obj_dict[bkt] = f"ha-obj{idx}-{t_t}"
+        LOGGER.info("Step 4: Copy object from %s to other buckets in background", bucket_name)
+        args = {'s3_test_obj': self.s3_test_obj, 'bucket_name': bucket_name, 'put_etag': put_etag,
+                'object_name': self.object_name, 'bkt_obj_dict': bkt_obj_dict, 'output': output,
+                'file_path': self.multipart_obj_path, 'background': True, 'bkt_op': bkt_op}
+        thread = threading.Thread(target=self.ha_obj.create_bucket_copy_obj, args=(event,),
+                                  kwargs=args)
+        thread.daemon = True  # Daemonize thread
+        thread.start()
+        LOGGER.info("Step 4: Successfully started background process for copy object")
+        # While loop to sync this operation with background thread to achieve expected scenario
+        if CMN_CFG["dtm0_disabled"]:
+            LOGGER.info("Waiting for creation of %s buckets", bkt_cnt)
+            bkt_list = list()
+            timeout = time.time() + HA_CFG["common_params"]["bucket_creation_delay"]
+            while len(bkt_list) < bkt_cnt:
+                time.sleep(HA_CFG["common_params"]["20sec_delay"])
+                bkt_list = self.s3_test_obj.bucket_list()[1]
+                if timeout < time.time():
+                    LOGGER.error("Bucket creation is taking longer than % sec",
+                                 HA_CFG["common_params"]["bucket_creation_delay"])
+                    assert_utils.assert_true(False, "Please check background process logs")
+        LOGGER.info("Waiting for %s sec...", HA_CFG["common_params"]["20sec_delay"])
```

Comment:
why we need this delay even if no bucket ops involved?

Reply:
Here we are waiting to perform some copy object operation in back ground, that's why we added wait